ncm-openstack/src/main/perl/OpenStack/Service.pm

```diff
@@ -534,18 +540,20 @@ Must return 1 on success.
 sub populate_service_database
 {
     my ($self) = @_;
+
+    # execute pre_populate commands before everything else
+    $self->pre_populate_service_database();
     # db_version is slow when not initialised
     # (lots of retries before it gives up; can take up to 90s)
-    if ($self->_do([$self->{manage}, @{$self->{db_version}}], 'determine database version', test => 1)) {
+    if ($self->_do(['/bin/bash', '-c', join(' ', $self->{manage}, @{$self->{db_version}})], 'determine database version', test => 1)) {
```

Comment:
@stdweird this is just a workaround to avoid the execution issue with `ceilometer-upgrade`. 
I have changed the unittests as well but now it is a bit messy, some commands are using `/bin/bash` workaround some of them  are not.
We should provide a better solution (maybe just adding a new flag into `CAF::Process` to use the "right" environment)

Reply:
I've created a CAF issue for this, could you fill it in with more details of what would be required to avoid this workaround?
https://github.com/quattor/CAF/issues/272